src/TLMessage/TLMessage/ServerMessages/UserSelf.php

```diff
@@ -34,18 +34,18 @@ public function getStatus()
     }
 
     /**
-     * @return UserProfilePhoto|null
+     * @return UserProfilePhoto|ChatPhoto|null
      */
     public function getPhoto()
     {
-        try {
-            $photo = $this->getTlMessage()->getNode('photo');
-
+        $photo = $this->getTlMessage()->getNode('photo');
+        if (TLServerMessage::checkType($photo, 'userProfilePhoto')) {
             return new UserProfilePhoto($photo);
-        }catch (TGException $e){
-            return null;
+        } elseif (TLServerMessage::checkType($photo, 'chatPhoto')) {
+            return new ChatPhoto($photo);
```

Comment:
Pls add test for this logic.
Save the binary you receive, decode it (see `OwnDeserializerTest` for example) and check the returned value.